hypershift-operator/metrics.go

```diff
@@ -117,6 +120,11 @@ func newMetrics(client crclient.Client, log logr.Logger) *hypershiftMetrics {
 			Name: "hypershift_nodepools_size",
 			Help: "Number of replicas associated with a given NodePool",
 		}, []string{"name", "platform"}),
+		// hypershiftOperatorInfo is a metric to capture the current operator details of the management cluster
+		hypershiftOperatorInfo: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Name: "hypershift_operator_info",
+			Help: "Indicates the details of the Hypershift Operator running in the management cluster.",
+		}, []string{"version", "image", "image_id", "go_version"}),
```

Comment:
The way this is, it won't expose the running operator but rather a time series with 1 for all operators in history? cc @muraee prove me wrong.

The values I'd like to capture are
```
 containerStatuses:
  - containerID: cri-o://091229606ec11499663f64dee856a3571e91ba86dee54c160b2b45b67e7f46a5
    image: image-registry.openshift-image-registry.svc:5000/hypershift/hypershift-operator@sha256:187e8368f60efa44406e147d90ad5faf5f97cfaebf2bf53a33685ce4ec428c38
    imageID: image-registry.openshift-image-registry.svc:5000/hypershift/hypershift-operator@sha256:187e8368f60efa44406e147d90ad5faf5f97cfaebf2bf53a33685ce4ec428c38
    lastState: {}
    ready: true
    restartCount: 0
```
```
runtime.Version()
supportedversion.LatestSupportedVersion?
```
and 
```
spec.image
```

I guess that'd need to be different metrics with GaugeValues?

Reply:
Agree, one thing to keep in mind is that previous values will always exist in the tsdb. One approach we could take is that instead of using "1" as the value, we use the timestamp that the operator last transitioned to "Available". The CVO does something similar in that it uses the timestamp that the version was completed to report its version. This way, for any given management cluster we can query for the max value (most recent) of this metric for a given cluster uuid.

Reply:
right, you would want to use `NewGaugeFunc` for constant values.
See https://github.com/prometheus/common/blob/main/version/info.go#L36-L56 for an example.

Reply:
@csrwng done, I've added a new function to handle the `NewGaugeFunc` update. The value reflects the seconds from the `containerStatuses.state.running.startedAt` until the `lastTransitionTime` in the `status.condition` (type Ready) field.